server/user/bitrot-detect

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+NCPUS=${1}
+if [[ -z "${NCPUS}" ]]; then
+    echo "Missing number of CPUs to use" >&2
+    exit 1
+fi
+
+TARGETDIR=${2}
+if [[ -z "${TARGETDIR}" ]]; then
+    echo "Missing target directory argument" >&2
+    exit 1
+fi
+
+WORKDIR=/var/tmp/bitrot-detect
+mkdir -p ${WORKDIR}
+
+# Clean up previous run.
+rm -f ${WORKDIR}/tbs.md5-*
+
+find ${TARGETDIR}/ -maxdepth 1 -type d -printf '%P\n' | sort | while read ctrl; do
+    cat ${TARGETDIR}/${ctrl}/*.md5 2> /dev/null | while read hash file; do
+        echo "${hash} ${TARGETDIR}/${ctrl}/${file}"
+    done
+done > ${WORKDIR}/tbs.md5 2> ${WORKDIR}/tbs.err
```

Comment:
Alternatively, perhaps something like this:
```
# Find, within the list of immediate subdirectories of the target directory, all files
# ending in ".md5"; transcribe their contents, replacing the filename with the full
# path (of the MD5 file minus the ".md5" extension, to the output; sort the output
# by path.
find $(find ${TARGETDIR} -maxdepth 1 -type d) \
    -maxdepth 1 -type f -name "*.md5" \
    -exec awk '{ sub(".md5", "", FILENAME); print $1, FILENAME }' {} \+ \
    2> ${WORKDIR}/tbs.err | sort -k2,2 >${WORKDIR}/tbs.md5
```

Reply:
See the latest code which adds some additional checks.